### PR TITLE
NFT Main module + logic

### PR DESCRIPTION
**Pull request by Viscount88**

### CODE REVIEW

**Review by DRadmir — CHANGES_REQUESTED** [review]

File: fearless/CoreLayer/OperationFactory/NFT/AlchemyNFTOperationFactory.swift [hunk 1]
```diff
@@ -142,12 +142,59 @@ final class AlchemyNFTOperationFactory {
...
+                    chain: chain
+                )
+
+                return NFT(
+                    chain: chain,
+                    tokenId: $0.id?.tokenId,
+                    title: $0.title,
+                    description: $0.description,
+                    smartContract: $0.contract?.address,
+                    metadata: nil,
```
[the remark this answers is not in the document]
Comment (Viscount88) [reply fearless/CoreLayer/OperationFactory/NFT/AlchemyNFTOperationFactory.swift:190]: fixed

File: fearless/ApplicationLayer/Services/NFT/Ethereum/AlchemyNftFetchingService.swift [hunk 2]
```diff
@@ -120,4 +137,16 @@ extension AlchemyNftFetchingService: NFTFetchingServiceProtocol {
 
         return collections
     }
+
+    func fetchCollectionNfts(collectionAddress: String, chain: ChainModel) async throws -> [NFT] {
+        let nfts = await withThrowingTaskGroup(of: [NFT]?.self) { [weak self] _ in
```
[the remark this answers is not in the document]
Comment (Viscount88) [reply fearless/ApplicationLayer/Services/NFT/Ethereum/AlchemyNftFetchingService.swift:188]: fixed

File: fearless/CoreLayer/OperationFactory/NFT/AlchemyNFTOperationFactory.swift [hunk 3]
```diff
@@ -142,12 +142,59 @@ final class AlchemyNFTOperationFactory {
         return operation
     }
 
+    private func createMapOwnedNftsOperation(
+        dependingOn remoteOperation: BaseOperation<AlchemyOwnedNftsResponse>,
```
[the remark this answers is not in the document]
Comment (Viscount88) [reply fearless/CoreLayer/OperationFactory/NFT/AlchemyNFTOperationFactory.swift:155]: discussed

File: fearless/Modules/NFT/MainNftContainer/MainNftContainerRouter.swift [hunk 4]
```diff
@@ -16,4 +16,6 @@ final class MainNftContainerRouter: MainNftContainerRouterInput {
 
         view?.controller.navigationController?.present(navigationController, animated: true)
     }
+
+    func showFilters(from _: ControllerBackedProtocol?) {}
```
[the remark this answers is not in the document]
Comment (Viscount88) [reply fearless/Modules/NFT/MainNftContainer/MainNftContainerRouter.swift]: fixed

**Pull request closed without merging**